client/components/week-progress-chart/week-chart.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { createRef } from 'react';
+import { scaleBand, scaleLinear, axisLeft, axisBottom, max, select, line } from 'd3';
+
+export default class WeekChart extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      SVGwidth: 0,
+      chartWidth: 0
+    };
+    this.SVGref = createRef();
+    this.DIVref = createRef();
+    this.drawChart = this.drawChart.bind(this);
+    this.resizeChart = this.resizeChart.bind(this);
+    this.dimensions = {
+      height: 250,
+      chartHeight: 225,
+      marginLeft: 32,
+      marginTop: 4
+    };
+  }
+
+  componentDidMount() {
+    const { DIVref, drawChart, resizeChart } = this;
+    this.setState({
+      SVGwidth: DIVref.current.offsetWidth,
+      chartWidth: DIVref.current.offsetWidth - 25
+    }, () => drawChart());
+    window.addEventListener('resize', resizeChart);
+  }
```

Comment:
Be sure to also include a `componentWillUnmount` to remove your resize listener from the window so we don't get a memory leak.